extensions/shared/external-media/constants.js

```diff
@@ -176,7 +176,7 @@ export const MONTH_SELECT_OPTIONS = [
 	{ label: __( 'Any Month', 'jetpack' ), value: -1 },
 	...map( range( 0, 12 ), value => ( {
 		// Following call generates a new date object for the particular month and gets its name.
-		label: dateI18n( 'F', new Date( [ 0, value + 1 ] ) ),
+		label: dateI18n( 'F', new Date( 0, value + 1 ) ),
```

Comment:
It seems that, at least in Chrome, `new Date( [ year, month ] )` takes a 1-based month. The standard `new Date( year, month )`, on the other hand, takes a 0-based month. So the `+ 1` here winds up labeling the months incorrectly in the generated dropdown.
```suggestion
		label: dateI18n( 'F', new Date( 0, value ) ),
```

Reply:
Yep, indeed:

![image](https://user-images.githubusercontent.com/96308/100013859-cfa65300-2dd5-11eb-97a9-b857ac446719.png)

Reply:
Thanks @anomiex! Applied in dd5b632.

Reply:
> It seems that, at least in Chromium, new Date( [ year, month ] ) takes a 1-based month. The standard new Date( year, month ), on the other hand, takes a 0-based month. So the + 1 here winds up labeling the months incorrectly in the generated dropdown.

TIL! Thanks for testing this out!